plugins/autodim.koplugin/main.lua

```diff
@@ -0,0 +1,257 @@
+--[[--
+Plugin for automatic dimming of the frontlight after an idle period.
+
+@module koplugin.autodim
+--]]--
+
+local Device = require("device")
+local Event = require("ui/event")
+local FFIUtil = require("ffi/util")
+local SpinWidget = require("ui/widget/spinwidget")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local time = require("ui/time")
+local _ = require("gettext")
+local T = FFIUtil.template
+
+local DEFAULT_AUTODIM_STARTTIME_M = 5
+local DEFAULT_AUTODIM_DURATION_S = 20
+local DEFAULT_AUTODIM_FRACTION = 20
+local AUTODIM_EVENT_FREQUENCY = 2 -- in Hz; Frequenzy for FrontlightChangedEvent on E-Ink devices
+
+local AutoDim = WidgetContainer:new{
+    name = "autodim",
+}
+
+function AutoDim:init()
+    self.autodim_starttime_m = G_reader_settings:readSetting("autodim_starttime_minutes", -1)
+    self.autodim_duration_s = G_reader_settings:readSetting("autodim_duration_seconds", DEFAULT_AUTODIM_DURATION_S)
+    self.autodim_fraction = G_reader_settings:readSetting("autodim_fraction", DEFAULT_AUTODIM_FRACTION)
+
+    self.last_action_time = UIManager:getElapsedTimeSinceBoot()
+
+    self.ui.menu:registerToMainMenu(self)
+    UIManager.event_hook:registerWidget("InputEvent", self)
+
+    self:_schedule_autodim_task()
+    self.isCurrentlyDimming = false -- true during or after the dimming ramp
+end
+
+function AutoDim:addToMainMenu(menu_items)
+    menu_items.autodim = self:getAutodimMenu()
+end
+
+function AutoDim:getAutodimMenu()
+    return {
+        text = _("Automatic dimmer"),
+        checked_func = function() return self.autodim_starttime_m > 0 end,
+        sub_item_table = {
+            {
+                text_func = function()
+                    return self.autodim_starttime_m <= 0 and _("Idle time for dimmer") or
+                    T(_("Idle time for dimmer: %1 min"), self.autodim_starttime_m)
+                end,
+                checked_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local idle_dialog = SpinWidget:new{
+                        title_text = _("Automatic dimmer idle time"),
+                        info_text = _("Start the dimmer after the designated period of inactivity."),
+                        value = self.autodim_starttime_m >=0 and self.autodim_starttime_m or 0.5,
+                        default_value = DEFAULT_AUTODIM_STARTTIME_M,
+                        value_min = 0.5,
+                        value_max = 60,
+                        value_step = 0.5,
+                        value_hold_step = 5,
+                        unit = _("min"),
```

Comment:
We mentionned using _m instead of _min for suffix of variables with duration in minutes.
Is it less confusing when it is "min" in the UI (which I could read a "minimal value" :)

Reply:
I slept over this. I think that using "min" (minute) as unit here is very precise and would _not_ lead any user into the wrong direction. (And we also have an info_text that a period of time has to be entered).
Especially as we have the unit "m" (meter) in autowarmth. And if we use in the UX everywhere "m" for minute, what shall we use for meter?

Reply:
For the user side of things min and m are both fine (neither confusing minute/minimum and meter/minute seems very likely?), but for the translator side of things these want to be context time vs context distance or some such.